### ISSUE TITLE
Server extension doesn't seem to be loading (jlab 3.0)

### ISSUE THREAD

**Issue by dialvarezs** [ryantam626/jupyterlab_code_formatter#184]
**Checklist prior to opening an issue**
- [X] I have followed fully the installation steps laid out in the documentation site.
- [X] I have restarted jupyterlab.
- [X] I have read the FAQ section in the documentation site.

**Describe the bug**
The ser extension doesn't seem to be loading in jupyter lab 3.0.

**Diagnostic commands**
When I open jupyter lab in the browser I get the error message "Unable to find server plugin version, this should be impossible,open a GitHub issue if you cannot figure this issue out yourself."
In the terminal I see the following lines regarding the extension (with `--debug`):
```
[D 2021-01-02 14:47:45.920 ServerApp] Path @ryantam626/jupyterlab_code_formatter/static/remoteEntry.5c93eb886a3b8e74175b.js served from /home/darkxhunter/.local/share/jupyter/labextensions/@ryantam626/jupyterlab_code_formatter/static/remoteEntry.5c93eb886a3b8e74175b.js
[D 2021-01-02 14:47:45.921 ServerApp] 200 GET /lab/extensions/@ryantam626/jupyterlab_code_formatter/static/remoteEntry.5c93eb886a3b8e74175b.js (::1) 1.21ms
[D 2021-01-02 14:47:46.329 ServerApp] Path @ryantam626/jupyterlab_code_formatter/static/160.8feff4fb8d6302eb02fa.js served from /home/darkxhunter/.local/share/jupyter/labextensions/@ryantam626/jupyterlab_code_formatter/static/160.8feff4fb8d6302eb02fa.js
[D 2021-01-02 14:47:46.329 ServerApp] 200 GET /lab/extensions/@ryantam626/jupyterlab_code_formatter/static/160.8feff4fb8d6302eb02fa.js (::1) 0.85ms
[W 2021-01-02 14:47:48.916 ServerApp] 404 GET /jupyterlab_code_formatter/version?1609609668137 (::1) 615.31ms referer=http://localhost:8888/lab
```

- `pip freeze | grep jupyter`
```
jupyter-client==6.1.7
jupyter-console==6.2.0
jupyter-core==4.7.0
jupyter-server==1.1.3
jupyterlab==3.0.0
jupyterlab-code-formatter==1.4.1
jupyterlab-pygments==0.1.2
jupyterlab-server==2.0.0
```

- `jupyter labextension list`
```
JupyterLab v3.0.0
/home/darkxhunter/.local/share/jupyter/labextensions
        @ryantam626/jupyterlab_code_formatter v1.4.1 enabled OK (python, jupyterlab-code-formatter)
```

- `jupyter serverextension list`
```
config dir: /home/darkxhunter/.jupyter
    jupyterlab_code_formatter  enabled 
    - Validating...
      X is jupyterlab_code_formatter importable?
```

**Additional context**
I'm using everything from user directory, not a virtualenv. Python 3.9 in OpenSUSE Tumbleweed.
I also completely cleaned the jupyter installation (removing `~/.jupyter`, `~/.local/share/jupyter` and `~/.local/etc/jupyter`, then reinstalling again jupyter and the extension), obtaining the same result.

Any ideas? Thanks!

**Comment by ryantam626** [comment]
Oh no 😭 I was hoping this would not happen.. 

The 
```
config dir: /home/darkxhunter/.jupyter
    jupyterlab_code_formatter  enabled 
    - Validating...
      X is jupyterlab_code_formatter importable?
```
thing *might* be artifact of a bug in jupyterlab right now. I got this but the plugin is working for me as normal otherwise.

I will need to try to replicate your setup to see if I can figure out what's wrong.

So am I correct in thinking you are using your system's Python (3.9) not within virtualenv/conda/pyenv?

**Comment by dialvarezs** [comment]
> So am I correct in thinking you are using your system's Python (3.9) not within virtualenv/conda/pyenv?

Yes, you are right.

I tried in a virtualenv and it works, so it could be something related the resolving the path maybe? I have not seen any other extensions using the new packaging, so you could be right about being a jlab bug.

**Comment by ryantam626** [comment]
Oh it works in a venv!? Okay that's a good lead, I got a spare laptop that I can experiment on, I might try replicating your setup tomorrow then

**Comment by lucasfigueiredo** [comment]
Hello, passing by to say that the below message appeared to me BUT the plug-in is working normally... :)

>     - Validating...
>       X is jupyterlab_code_formatter importable?

**Comment by kerwinyc** [comment]
> Oh it works in a venv!? Okay that's a good lead, I got a spare laptop that I can experiment on, I might try replicating your setup tomorrow then

Great ryantam ,  I installed in a virtualenv,  jupyterlab3.0 .but got this messages.   anything I can do to fix it ? Thanks a lot.

Jupyterlab Code Formatter Error
Unable to find server plugin version, this should be impossible,open a GitHub issue if you cannot figure this issue out yourself.

@ryantam626/jupyterlab_code_formatter
This is a prebuilt extension. To uninstall it, please apply following instructions.
Use your Python package manager (pip, conda, etc.) to uninstall the package jupyterlab-code-formatter

**Comment by cbertolasio** [comment]
I also tried installing v1.4.1 of the extension into jupyterlab v 3.0.1.

I get the same output when running jupyter serverextension list:
 X is jupyterlab_code_formatter importable?
 
 My particular setup is a vanilla azure machine learning vm that has jupyterlab built in with python 3.6.

The plugin shows up in the advanced settings editor but none of the commands are available in the command pallet.

**Comment by ryantam626** [comment]
@dialvarezs I didn't manage to revive my laptop, so can't replicate your setup exactly. I did run some test with docker instead, I think there might be something wrong with your Python setup.
Here is what I did,
1. Get a shell with your OS
```
docker run -it -p 8888:8888 opensuse/tumbleweed bash
```
2. Run these
```
zypper in python3
zypper in python3-pip
pip3 install jupyterlab black isort jupyterlab_code_formatter
jupyter serverextension enable --py jupyterlab_code_formatter
jupyter lab --allow-root --ip 0.0.0.0
```

The plugin seems to be working in this test.

**Comment by ryantam626** [comment]
@kerwinyc how did you install the plugin?

**Comment by ryantam626** [comment]
@cbertolasio can you confirm you have one of the formatters (e.g. black/isort/autopep8) installed? And also were you trying to use the plugin with a notebook/python file opened?

**Comment by dialvarezs** [comment]
@ryantam626 I tried in a VM and it happens as well. Worried about being something specific about Tumbleweed I tried with a Fedora 33 VM, same thing. I think this doesn't happen in a container because there everything run as root using system directories, it's not the same as a user install.
What I execute exactly is
```
pip3 install --user jupyterlab black isort jupyterlab_code_formatter
jupyter serverextension enable --py jupyterlab_code_formatter
jupyter lab
```
(I also tried `--user` in the extension enabling but it's the same)
So I think it is something specific for the user install (because it works for system install and also virtualenv).

Let me know if I can help you in some way with the debugging.

**Comment by cbertolasio** [comment]
> n with a notebook/python file opened

yep, i was trying to reference format commands from the command pallet in a notebook.  i had originally installed via pip.  and my versions of black are 20.8b1, and isort 5.5.4,

i uninstalled and reinstalled via pip, and I was still getting error outputin jupyter serverextension list, so, I installed via conda, restarted jupyterlab and things started working.  I now have a format icon in the notebook, and if i right click on a cell i get the format option.

so, i am guessing my issue is fixed.

**Comment by dialvarezs** [comment]
I was able to getting it work finally!
With jupyterlab 3, the command to enable the extension is
```
jupyter server extension enable jupyterlab_code_formatter
```
With the space between "server" and "extension". Using `serverextension` enables the extension using the "old" way (creating `jupyter_notebook_config.json`), with the new one a json file is added to `jupyter_server_config.d` ([link](https://jupyter-server.readthedocs.io/en/latest/operators/multiple-extensions.html)). In my case was neccesary to add the `--user` option. A tricky one...

Executing `jupyter server extension list`, the output is
```
Config dir: /home/darkxhunter/.jupyter
    jupyterlab_code_formatter enabled
    - Validating jupyterlab_code_formatter...
      jupyterlab_code_formatter 1.4.1 OK

Config dir: /usr/etc/jupyter

Config dir: /usr/local/etc/jupyter
```

And everything is working as expected.

I also noticed in the documentation that in `__init__.py` the function to make the extension discoverable should be called `_jupyter_server_extension_points` ([link](https://jupyter-server.readthedocs.io/en/latest/developers/extensions.html#making-an-extension-discoverable)), but it seems to work with `_paths` for now, so it shouldn't be a problem, specially if it breaks the instalation in jlab < 3.0.0.
